### ISSUE TITLE
'z' in skeleton breaks date formatting

### ISSUE THREAD

**Issue by oliver-stripe** [globalizejs/globalize#767]
Attempting to include timezone with `z` seems to break date formatting. Example (last console.log seems incorrect):

```
var Globalize = require('globalize');
Globalize.load(require('cldr-data').entireSupplemental());
Globalize.load(require('cldr-data').entireMainFor('en'));
Globalize.loadTimeZone(require('iana-tz-data'));

console.log(
  Globalize('en').formatDate(new Date(), {datetime: 'full', timeZone: 'America/Los_Angeles'})
); // Friday, August 11, 2017 at 10:21:52 PM Pacific Daylight Time

console.log(
  Globalize('en').formatDate(new Date(), {skeleton: 'yMMMMdhmmv', timeZone: 'America/Los_Angeles'})
); // August 11, 2017 at 10:22 PM PT

console.log(
  Globalize('en').formatDate(new Date(), {raw: 'MMMM d, y, h:mm z', timeZone: 'America/Los_Angeles'})
); // August 11, 2017, 10:22 PDT

console.log(
  Globalize('en').formatDate(new Date(), {skeleton: 'yMMMMdhmmz', timeZone: 'America/Los_Angeles'})
); // August 11, 2017
```

The last output seems like it should be `August 11, 2017 at 10:22 PM PDT`, or am I doing something wrong?

**Comment by rxaviers** [comment]
You are correct, you should be able to use `{skeleton: 'yMMMMdhmmz'}` just fine. This requires one enhancement in globalize, specifically the ability for the [getBestMatchPattern](https://github.com/globalizejs/globalize/blob/master/src/date/expand-pattern/get-best-match-pattern.js) to handle **appendItems**:

> In case the best match does not include all the requested calendar fields, the appendItems element describes how to append needed fields to one of the existing formats. Each appendItem element covers a single calendar field. In the pattern, {0} represents the format string, {1} the data content of the field, and {2} the display name of the field (see Calendar Fields).

Useful links:
- http://www.unicode.org/reports/tr35/tr35-dates.html#availableFormats_appendItems
- https://github.com/unicode-cldr/cldr-dates-modern/blob/master/main/en/ca-gregorian.json#L384-L396

This suport would be very welcome.

**Comment by abhisheksreepal** [comment]
@rxaviers  Also following input formats produces following incorrect output. 

Input
Globalize.formatDate( new Date(), {
      skeleton: "yMMMdhmVVVV",
      timeZone: Intl.DateTimeFormat().resolvedOptions().timeZone
    });


Actual Output:
Jun 23, 2019

Expected Output:
Jun 23, 2019 1:30 PM India Standard Time
